### PR TITLE
chore: pyupgrade 3.9+

### PR DESCRIPTION
**Pull request by henryiii**
### Description

This enables pyupgrade and upgrades older syntax to 3.9+ syntax.

Normally I'd use isort's include injection to add `from __future__ import annotations`, which provides a small performance benefit and a large readabilty benefit for codebases targeting older Python version, but a) the includes are not sorted by isort (would you like that) so this would sort _and_ inject, and b) with 3.9+, you actually get some nice readablity on typing already with the default generics - the main missing one is `Union[A, B]` -> `A | B` (which would come if/when this goes to 3.10+ as well). So this just focus on pyupgrade.

I forgot to exclude the micropip pip vendor, should that be excluded?

Also added a manual pre-commit way to clean up unused imports, which saved some time.

### Checklists

- [ ] Add a [CHANGELOG](https://github.com/pyodide/pyodide/blob/main/docs/project/changelog.md) entry
- [ ] Add / update tests
- [ ] Add new / update outdated documentation

### CODE REVIEW

**Comment by hoodmane** [comment]
> the includes are not sorted by isort (would you like that)

I personally would like that. Normalizing the include order removes one pointless thing to worry about.

**Comment by henryiii** [comment]
Okay, can do that in a follow up, and I agree. Reduces potential merge conflicts too, the same addition is always in the same place.

File: src/tests/test_console.py [hunk 1]
```diff
@@ -429,7 +429,7 @@ def __repr__(self):
 
             >>> Test()
             [[;;;terminal-error]Traceback (most recent call last):
-              File \"/lib/python3.9/site-packages/_pyodide/console.py\", line 478, in repr_shorten
+              File \"/lib/python3.9/site-packages/_pyodide/console.py\", line 475, in repr_shorten
                 text = repr(value)
```
Comment (hoodmane) [inline src/tests/test_console.py:433]: We really ought to remove this number from the string before making the comparison...

**Comment by ryanking13** [comment]
Thanks, @henryiii!

> I forgot to exclude the micropip pip vendor, should that be excluded?

Yes, I think these should better be excluded.

> Normally I'd use isort's include injection to add `from __future__ import annotations`

If this line only affects Python versions < 3.10, I would prefer not to add it right now. Because we (hopefully) will upgrade Pyodide's Python version before long, and we don't maintain older Python compatibility.
Apart from that, sorting import orders with `isort` would be great.

File: src/tests/test_console.py (hunk 1 above)
Comment (ryanking13) [reply src/tests/test_console.py:433]: Yeah...

File: pyodide-build/pyodide_build/buildpkg.py [hunk 2]
```diff
@@ -633,7 +633,7 @@ def build_package(
         return
 
     if not should_prepare_source and not srcpath.exists():
-        raise IOError(
+        raise OSError(
```
Comment (hoodmane) [inline pyodide-build/pyodide_build/buildpkg.py:636]: Why did `IOError` get replaced with `OSError`? Is `IOError` deprecated?

**Review by hoodmane — APPROVED** [review]

File: pyodide-build/pyodide_build/buildpkg.py (hunk 2 above)
Comment (hoodmane) [reply pyodide-build/pyodide_build/buildpkg.py:636]: Oh:

> Changed in version 3.3: [EnvironmentError](https://docs.python.org/3/library/exceptions.html#EnvironmentError), [IOError](https://docs.python.org/3/library/exceptions.html#IOError), [WindowsError](https://docs.python.org/3/library/exceptions.html#WindowsError), [socket.error](https://docs.python.org/3/library/socket.html#socket.error), [select.error](https://docs.python.org/3/library/select.html#select.error) and mmap.error have been merged into [OSError](https://docs.python.org/3/library/exceptions.html#OSError), and the constructor may return a subclass.

Comment (hoodmane) [reply pyodide-build/pyodide_build/buildpkg.py:636]:
```py
>>> IOError is OSError
True
```

**Comment by hoodmane** [comment]
Thanks again @henryiii for your maintenance help. We appreciate it a lot.

**Pull request merged**